Refactor Staff model

adevade opened the pull request:
**Note:** This PR is easiest to review commit by commit. Feel free to pick and choose what you like and revert the ones you don't.

The reason for removing the Docblocks is because Filament barely uses any. With clear property/method names, type hints and return types they often feel superfluous. 

I _think_ that we can remove the custom route binding after [this PR was merged in Livewire](https://github.com/livewire/livewire/pull/7913)(?) 

I usually use my accessors as the rest of my attributes (snake cased), but don't know if you have a strong preference? Both still works. Especially in `$appends` I think it should be snake cased, since otherwise it would be mixing case with the other fields from the DB.

Marking as draft for now, to let the tests run and get some feedback.

adevade committed:
Reorder properties and methods

adevade committed:
Remove custom route model binding

https://github.com/livewire/livewire/pull/7913

adevade committed:
Remove docblocks

adevade committed:
Add missing cast

adevade committed:
Update `full_name` accessor

adevade committed:
Update local search scope

adevade committed:
Update imports

adevade committed:
Use snake case accessor for consistency

adevade committed:
Add class docblock for IDEs

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

alecritson committed:
Merge branch '1.x' into refactor-staff-model

alecritson committed:
Tests

alecritson commented:
@adevade Sorry this one took a while, looks pretty okay to me. I've added a small test (which didn't exist before) lmk your thoughts and we can get this merged.

adevade commented:
@alecritson Nice work with the test! 👍

One thing I noticed is that `firstname` and `lastname` is defined without underscores. Should `fullname` be that as well, for consistency? Easier than migrating both first and last name columns I guess?

alecritson commented:
> @alecritson Nice work with the test! 👍
> 
> One thing I noticed is that `firstname` and `lastname` is defined without underscores. Should `fullname` be that as well, for consistency? Easier than migrating both first and last name columns I guess?

I think if anything we'd be better to migrate `firstname` to `first_name` it's something that's bothered us internally since elsewhere we use underscores. Not sure if it's something we need to worry about right now though @glennjacobs ?

adevade committed:
Merge branch '1.x' into refactor-staff-model

glennjacobs commented:
> > @alecritson Nice work with the test! 👍
> > One thing I noticed is that `firstname` and `lastname` is defined without underscores. Should `fullname` be that as well, for consistency? Easier than migrating both first and last name columns I guess?
> 
> I think if anything we'd be better to migrate `firstname` to `first_name` it's something that's bothered us internally since elsewhere we use underscores. Not sure if it's something we need to worry about right now though @glennjacobs ?

I guess if we don't do it now, then it may never get another look. So maybe we get it done in this PR?

adevade committed:
Merge branch '1.x' into refactor-staff-model

actions-user committed:
chore: fix code style

adevade committed:
Update `firstname` `lastname` to have underscores

adevade committed:
Merge branch 'refactor-staff-model' of https://github.com/adevade/lunarphp-lunar into pr/1793

adevade committed:
Fix migrations for SQLite

"SQLite doesn't support multiple calls to dropColumn / renameColumn in a single modification."

adevade committed:
Add fallbacks for old keys

adevade committed:
Fix fallbacks to actually work...

adevade commented:
I've now updated the code base to use `first_name` and `last_name`.

Since users might be referencing `$staff->firstname` in their own code, I've added fallback accessor and mutators for both `$firstname` and `$lastname` on the Staff model.

I did not update [the CLI argument keys on `MakeLunarAdminCommand`](https://github.com/lunarphp/lunar/blob/736d14c86f96be08d90a49643bac97f77aa2231f/packages/admin/src/Console/Commands/MakeLunarAdminCommand.php#L19-L20), since it feels better for them to not have underscores.

I have not yet touched any documentation.

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade committed:
Merge branch '1.x' into refactor-staff-model

adevade commented:
@alecritson Should still be good to merge 👍 The tests are passing. Haven't looked at it in a good while though...

adevade committed:
Merge branch '1.x' into refactor-staff-model
